rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -185,16 +185,26 @@ local function route_to_alternative_balancer(balancer)
   local target_header = util.replace_special_char(traffic_shaping_policy.header,
                                                   "-", "_")
   local header = ngx.var["http_" .. target_header]
+  local match_strategy = "exact"
+  if traffic_shaping_policy.matchStrategy and #traffic_shaping_policy.matchStrategy > 0 then
+    match_strategy = traffic_shaping_policy.matchStrategy
+  end
   if header then
     if traffic_shaping_policy.headerValue
-       and #traffic_shaping_policy.headerValue > 0 then
-      if traffic_shaping_policy.headerValue == header then
-        return true
+	   and #traffic_shaping_policy.headerValue > 0 then
+      if match_strategy == "exact"  then
+        if traffic_shaping_policy.headerValue == header then
+          return true
+        end
+      elseif  match_strategy == "regex" then
+          if string.match(header, traffic_shaping_policy.headerValue) then
```

Comment:
Should we use `ngx.re.match` to provide full PCRE Regex support? `string.match` does only Lua string matching which is limited. Maybe we should just stick to `string.match` to keep it simple, but please make sure this is documented, as match-strategy=regex might confuse people to think it will support full regex.

Reply:
I prefer ngx.re.match. 
done